### ISSUE TITLE
Native alsa support

### ISSUE THREAD

**Issue by davidheitman** [yuzu-emu/yuzu#10513]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What feature are you suggesting?

I know this is a long shot, there are many higher priorities for the project,but I wanted to request native alsa support under linux to allow for surround sound in yuzu without pulseaudio/cubeb.

### Why would this feature be useful?

To use surround sound with cubeb requires pulseaudio. Pulse can't handle DTS-MA passthrough, which is a requirement for me on my media center box, so I can't use it. Which means no surround sound in Yuzu (without config changes and reboots and such). This forces me to choose between pulse and lower sound quality in kodi, or alsa and no surround in yuzu. Not ideal, but I'd rather have 4k video in yuzu with no surround than sacrifice dts-ma in kodi.

For reference, this relates to my short-lived issue here: #10512

**Comment by davidheitman** [comment]
You guys are clearly on tonight, and I know it's not to protocol, but I thought I'd try my luck with an old fashioned bump. Any thoughts on this? I'd love to work on it, but time permitting, and time never permits.

**Comment by v1993** [comment]
After looking at the related issue and googling around - are you sure pipewire does not support TrueHD? Take a look at https://gitlab.freedesktop.org/pipewire/pipewire/-/issues/1895. It might require a bit of setup, but ultimately should work.

Regardless - on yuzu side of things it might end up getting merged if PRd, but realistically speaking, it's an *extremely* niche usecase that's better solved in pipewire.

Finally - it looks like SDL has a [native ALSA driver](https://github.com/libsdl-org/SDL/tree/main/src/audio/alsa), with quick look suggesting that it has support for surround sound. Maybe try selecting it as the audio backend in yuzu instead of cubeb?

**Comment by davidheitman** [comment]
Nah, pipewire has much the same limitations as pulse, DTS is doable, but DTS-MA/HD is a no go. The link you posted, guy got DTS working, but no MA, same as me.  Some years back pulse announced support for it in a release, but it was broken,got reverted, and not a peep since. And SDL is a no go from my testing, couldn't get surround even with the config enabled, and haven't seen any reports of it working for anyone else. Although it has been a while, maybe I'll play with it some more.

Well, if I ever get some time, I'll see if I can do anything. Maybe crib off dolphin's alsa backend for some example code. Thanks for looking though!
